.github/workflows/build.sh

```diff
@@ -20,7 +24,37 @@ case "$1" in
             export UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1
         fi
 
-        ./bootstrap.sh --enable-tests --prefix=/usr
+        if [[ "$WERROR" == true ]]; then
+            # acx_pthread isn't compatible with CFLAGS=-Werror at all so it's just replaced
+            # with ax_pthread from autoconf-archive.
+            rm -rf common/acx_pthread.m4
+
+            export CFLAGS="-Werror"
+
+            # The following warnings are ignored because it's hard to fix them all in one fell swoop.
+            # Ideally they should be fixed one by one and removed from here to let compilers catch them
+            # automatically on PRs.
+            if [[ "$CC" == gcc ]]; then
+                CFLAGS+=" -Wno-error=format-truncation -Wno-error=deprecated-declarations -Wno-error=implicit-fallthrough"
+            elif [[ "$CC" == clang ]]; then
+                CFLAGS+=" -Wno-error=strict-prototypes -Wno-error=invalid-source-encoding -Wno-error=varargs -Wno-error=cast-align"
+                CFLAGS+=" -Wno-error=logical-not-parentheses -Wno-error=deprecated-declarations"
+            fi
+
+            # avahi-gobject triggers a lot of various warnings so it's just easier to turn
+            # it off entirely instead of disabling all the warnings with -Wno-error.
+            additional_autogen_args+=" --disable-gobject"
+
+            # For some reason "howl" and "libdns_sd" completely ignore -Wno-error= passed via
+            # CFLAGS so they are ignored until they are fixed somehow.
+            additional_autogen_args+=" --disable-compat-howl --disable-compat-libdns_sd"
```

Comment:
Interesting. That should be fixed, as Fedora requires CFLAGS passing from rpm to builds. Haven't noticed this is not passed everywhere. May I ask separate issue for it too?

Reply:
I think I got it to work a few hours ago. Let me double-check.

Reply:
I brought those compat libraries back. I double-checked and it seems CFLAGS aren't ignored there. I'm not sure why they failed to build though. My guess would be that I missed a couple of flags to really silence all the warnings there.